lib/routes/account.js

```diff
@@ -1044,11 +1009,30 @@ module.exports = function (
         var sessionToken = request.auth.credentials
         var uid = sessionToken.uid
         db.devices(uid).then(
-          function (devices) {
-            reply(devices.map(function (device) {
+          function (deviceArray) {
+            reply(deviceArray.map(function (device) {
+              if (! device.name) {
+                device.name = devices.synthesizeName(device)
+                log.info('deviceName:synthesize', {
+                  uid: uid,
+                  id: device.id,
+                  name: device.name
+                })
+              }
+
+              if (! device.type) {
+                device.type = device.uaDeviceType || 'desktop'
+                log.info('deviceType:synthesize', {
```

Comment:
Ditto my previous comment about `log.info` API.  However, I'm not sure this is the best place to log an event for it.  IIUC the thing we want to know is, how many placeholder records are being created over time?  It may be simpler to just log the event at the time when we *create* the placeholder record, rather than logging an event every time we *read* a placeholder record.